src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/mappers/DateMapper.java

```diff
@@ -33,4 +33,15 @@ public static int secondsSinceStartOfTime(ZonedDateTime startOfTime, Instant ins
     public static LocalDateTime asDateTime(LocalDate localDate, int secondsSinceStartOfDay) {
         return localDate.atStartOfDay().plusSeconds(secondsSinceStartOfDay);
     }
+
+    public static int secondsSinceStartOfService(
+        ZonedDateTime departureDate, ZonedDateTime dateTime, ZoneId zoneId
+    ) {
+        ZonedDateTime startOfService = asStartOfService(departureDate, zoneId);
+        return (int) Duration.between(startOfService, dateTime).toSeconds();
+    }
+
+    private static ZonedDateTime asStartOfService(ZonedDateTime dateTime, ZoneId zoneId) {
+        return asStartOfService(dateTime.withZoneSameInstant(zoneId));
+    }
```

Comment:
I do not think this is correct. I had to write a little test to verify it, but
```
ZoneId fi = ZoneId.of("Europe/Helsinki");
ZoneId no = ZoneId.of("Europe/Oslo");
ZonedDateTime t0 = ZonedDateTime.of(2020, 11, 1, 0, 20, 0,0, fi);
ZonedDateTime t1 = ZonedDateTime.of(2020, 11, 1, 1, 20, 0,0, fi);
System.out.println(TimeUtils.timeToStrShort(secondsSinceStartOfService(t0, t0, no)));
System.out.println(TimeUtils.timeToStrShort(secondsSinceStartOfService(t1, t1, no)));
$ 23:20
$ 00:20
```
The Netex import uses the "raw" local date of the xsd:Date passed in as the service date. The `SiriTimetableSnapshotSource` seems to use the first estimated call and the "raw" date from that. 

One problem here is that for input times with time-zone ahead of the _graph timezone_ we may get negative time offset, assuming we use the "raw" date. The other option is to move the service date to the day before, but that is very confusing. So, I am in favor of using the "raw" date and then risk negative times.

Reply:
Changing line 40 to will use the "raw" date:
`ZonedDateTime startOfService = asStartOfService(departureDate.toLocalDate(), zoneId);`

Then the last method, line 44, can be deleted.

Reply:
As I commented on #3234 currently one unit test fails depending on system timezone.

For offsets of GMT-1, UTC, GMT+xx the test passes (`TZ=Etc/GMT±XX`), e.g.
```
TZ=Europe/Oslo LANG=C mvn -Dtest=SiriAlertsUpdateHandlerTest\#testSiriSxUpdateForTrip test
```
but for offsets of GMT-12 to GMT-2 the test fails, e.g.
```
TZ=Europe/Helsinki LANG=C mvn -Dtest=SiriAlertsUpdateHandlerTest\#testSiriSxUpdateForTrip test
```

At least this suggested change is not enough to fix the unit test, but is it actually that unit test somehow misusing system locale instead of RT update code?